test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.ValidateTokenAsyncTests.Common.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.TestUtils;
+using Microsoft.IdentityModel.Tokens.Saml2;
+
+namespace Microsoft.IdentityModel.Tokens.Saml.Tests
+{
+#nullable enable
+    public partial class Saml2SecurityTokenHandlerTests
+    {
+        internal static async Task ValidateAndCompareResults(
+            Saml2SecurityToken saml2Token,
+            TokenValidationParameters tokenValidationParameters,
+            ValidateTokenAsyncBaseTheoryData theoryData,
+            CompareContext context)
+        {
+            Saml2SecurityTokenHandler saml2TokenHandler = new Saml2SecurityTokenHandler();
+
+            // Validate the token using TokenValidationParameters
+            TokenValidationResult legacyTokenValidationParametersResult =
+                await saml2TokenHandler.ValidateTokenAsync(saml2Token.Assertion.CanonicalString, tokenValidationParameters);
+
+            // Validate the token using ValidationParameters
+            ValidationResult<ValidatedToken> validationParametersResult =
+                await saml2TokenHandler.ValidateTokenAsync(
+                    saml2Token,
+                    theoryData.ValidationParameters!,
+                    theoryData.CallContext,
+                    CancellationToken.None);
+
+            // Ensure the validity of the results match the expected result
+            if (legacyTokenValidationParametersResult.IsValid != theoryData.ExpectedIsValid)
+                context.AddDiff($"tokenValidationParametersResult.IsValid != theoryData.ExpectedIsValid");
+
+            if (validationParametersResult.IsSuccess != theoryData.ExpectedIsValid)
+                context.AddDiff($"validationParametersResult.IsSuccess != theoryData.ExpectedIsValid");
+
+            if (theoryData.ExpectedIsValid &&
+                legacyTokenValidationParametersResult.IsValid &&
+                validationParametersResult.IsSuccess)
+            {
+                // This should compare the ClaimsPrincipal and ClaimsIdentity from one result against the other but right now we have not defined how we will handle this
```

Comment:
please open an issue for tracking and link here

Reply:
Created an issue for tracking [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2881)